doc/cookbook/pipeline.md

```diff
@@ -0,0 +1,92 @@
+# Pipeline 
+
+- [Pipeline](#pipeline)
+  - [Background](#background)
+  - [Examples](#examples)
+    - [Sequences executing](#sequences-executing)
+    - [JumpIf](#jumpif)
+  - [References](#references)
+
+
+## Background
+
+* Easegress provides many useful filters, such as rate limiter, validator, request adaptor, and so on.
+* The `Pipeline` is an Easegress build-in mechanism for orchestrating HTTP filters for handling requests. It uses the `Chain of Responsibility Pattern`[1] and also supports conditional forward direction jumping. 
+* The `HTTPServer` receives incoming traffic and routes to one dedicated `Pipeline` in Easegress according to the HTTP header, path matching, etc.
+
+
+## Examples 
+
+### Sequences executing
+
+* The basic model of Pipeline execution is a sequence. Filters will be executed step by step in the order described by the `flow` field in Pipeline's spec.  
+
+```bash
+$ echo '
+name: pipeline-demo
+kind: HTTPPipeline
+flow:
+  - filter: requestAdaptor
+  - filter: proxy
+filters:
+  - name: requestAdaptor
+    kind: RequestAdaptor
+    header:
+      set:
+        X-Adapt-Key: goodplan
+  - name: proxy
+    kind: Proxy
+    mainPool:
+      servers:
+      - url: http://127.0.0.1:9095
+      - url: http://127.0.0.1:9096
+      - url: http://127.0.0.1:9097
+      loadBalance:
+        policy: roundRobin' | egctl object create
+```
+* The `pipeline-demo` above will execute `rateLimiter` filter first, then `proxy`. So the `proxy` filter can forward the request with the header `X-Adapt-Key` setting by `requestAdaptor`.  
+
+### JumpIf
+
+* Easegress' filter supports return execution result string message. If it's empty, that means these filters handle the request without failure. Otherwise, Easegress will   
```

Comment:
```suggestion
* Easegress' filter returns a string message as execution result. If it's empty, that means these filters handle the request without failure. Otherwise, Easegress will   
```

And this sentence seems incomplete.

Reply:
Updated